controllers/release/adapter.go

```diff
@@ -337,28 +353,40 @@ func (a *adapter) EnsureReleaseProcessingIsTracked() (controller.OperationResult
 	return controller.ContinueProcessing()
 }
 
-// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
-// Processing step are cleaned up once processing is finished.
-func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if !a.release.HasProcessingFinished() {
+// EnsureReleaseManagedProcessingIsTracked is an operation that will ensure that the Release Managed PipelineRun status
+// is tracked in the Release being processed.
+func (a *adapter) EnsureReleaseManagedProcessingIsTracked() (controller.OperationResult, error) {
+	if !a.release.IsManagedPipelineProcessing() || a.release.HasManagedPipelineProcessingFinished() {
 		return controller.ContinueProcessing()
 	}
 
-	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
-	if err != nil && !errors.IsNotFound(err) {
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release, metadata.ManagedPipelineType)
+	if err != nil {
 		return controller.RequeueWithError(err)
 	}
+	if pipelineRun != nil {
+		err = a.registerManagedProcessingStatus(pipelineRun)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+	}
 
-	roleBinding, err := a.loader.GetRoleBindingFromReleaseStatus(a.ctx, a.client, a.release)
-	if err != nil && !errors.IsNotFound(err) && !strings.Contains(err.Error(), "valid reference to a RoleBinding") {
-		return controller.RequeueWithError(err)
+	return controller.ContinueProcessing()
+}
+
+// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
+// Processing step are cleaned up once processing is finished. This exists in conjunction with EnsureFinalizersAreCalled because
+// the finalizers should be removed from the pipelineRuns even if the Release is not marked for deletion for quota reasons.
+func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
+	if !a.release.HasTenantPipelineProcessingFinished() || !a.release.HasManagedPipelineProcessingFinished() {
+		return controller.ContinueProcessing()
 	}
 
-	return controller.RequeueOnErrorOrContinue(a.cleanupProcessingResources(pipelineRun, roleBinding))
+	return controller.RequeueOnErrorOrContinue(a.finalizeRelease(false))
```

Comment:
Why not to delete them here as well?

Reply:
2 reasons. The first is that it is nice to have them around before they are pruned for debugging and not rely on tekton results. The second is that when I tried this, I was having issues with the tracking information (fwiw, this is how we currently have things. Clean up removes finalizers, finalizeRelease deletes it)